router/pkg/routerplugin/grpc_plugin.go

```diff
@@ -120,43 +112,92 @@ func (s *GRPCPlugin) Start(ctx context.Context, logger *zap.Logger) error {
 		return fmt.Errorf("plugin does not implement grpc.ClientConnInterface")
 	}
 
-	s.cp, err = newGRPCPluginClient(pluginClient, grpcClient)
-	if err != nil {
-		return fmt.Errorf("failed to create grpc plugin client: %w", err)
+	if p.client == nil {
+		// first time we start the plugin, we need to create a new client
+		p.client, err = newGRPCPluginClient(pluginClient, grpcClient)
+		if err != nil {
+			return fmt.Errorf("failed to create grpc plugin client: %w", err)
+		}
+		return nil
+	}
+
+	p.client.setClients(pluginClient, grpcClient)
+
+	return nil
+
+}
+
+// Name implements Plugin.
+func (p *GRPCPlugin) Name() string {
+	return p.pluginName
+}
+
+// Start implements Plugin.
+func (p *GRPCPlugin) Start(ctx context.Context, logger *zap.Logger) error {
+	if logger == nil {
+		logger = zap.NewNop()
+	}
```

Comment:
When should this argument be allowed to be nil and not be programmer error? Shouldn't they pass zap.NewNop at the call site if this is the desired behaviour?